packages/vscode-extension/src/debug/npmLogHandler.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { spawn } from "child_process";
+import * as fs from "fs-extra";
+import * as path from "path";
+
+function getNpmCachePath(): Promise<string> {
+  const command = spawn("npm config get cache", {
+    shell: true,
+  });
+  return new Promise((resolve, reject) => {
+    command.stdout.on("data", (data) => {
+      resolve(`${data.toString().trim()}`);
```

Comment:
stdout.on("data") returns whenever the process write to stdout. This does not mean the process exits. You should append the result returned from stdout.on("data") to a buffer and use the buffer when command.on("exit") or command.on("error"). see [cputils code](https://github.com/OfficeDev/TeamsFx/blob/f841ea22ad6b66c6f8cebbbff8b7d12de6949c32/packages/vscode-extension/src/debug/depsChecker/cpUtils.ts#L64)

Reply:
In normal cases, `npm config get cache` only has one line output. I wonder when will this command have more than one line output.

Reply:
This has nothing todo with how many lines it returns.

If the node process run this:
```
process.stdout.write("/home");
process.stdout.write("/alexwang");
process.stdout.write("/.npm");
```

You will get 3 `data` event.

Or more common cases, if node use `process.stdout.write("/home/alexwang/.npm")` to output but the underlying buffer of stdin is full when it writes only part of the string, node or the underlying libc will write the first part of the string to stdout, which will trigger you `data` event.

Anyway, I think we should not depend on the implementation details and you can reuse cpUtils to prevent this.

Reply:
Make sense.

Reply:
Updated to use `cpUtils.executeCommand`.